src/modules/navigator/mission_block.cpp

```diff
@@ -1011,3 +1011,31 @@ void MissionBlock::startPrecLand(uint16_t land_precision)
 		_navigator->get_precland()->on_activation();
 	}
 }
+
+void MissionBlock::updateAltToAvoidTerrainCollisionAndRepublishTriplet(mission_item_s mission_item)
+{
+	// Avoid flying into terrain using the distance sensor. Enable through the parameter NAV_MIN_GND_DIST.
+	// Only active during commanded descents with vz>0 (to prevent climb-aways), excluding landing and VTOL transitions.
+	// It changes the altitude setpoint in the triplet to maintain the current altitude and republish the triplet.
+	// We also change the mission item altitude used for acceptance calculations to prevent getting stuck in a loop.
+
+	const float kAltitudeDifferenceForDescentCondition = 5.f; // 5 meters threshold
```

Comment:
```suggestion
	static constexpr float kAltitudeDifferenceForDescentCondition = 5.f; // meters threshold
```
So from 4 meters altitude it's still fine to fly into the ground? This is probably only there to make sure it doesn't trigger (again) if the position estimate is 10cm higher than the triplet (that was already adjusted) 🤔

Reply:
Yes correct, that was the reason I put it there, and yes, that means if the vehicle is less than 5m vertical distance away from the setpoint it wouldn't trigger the collision prevention. Very random number, I agree. 
I've reduced it to 2m to mitigate it slightly. Otherwise, to get rid of the magic threshold, we could only allow the terrain correction once per set triplet (and reset if a new triplet comes in).